test/e2e/serial/tests/metrics.go

```diff
@@ -0,0 +1,94 @@
+package tests
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	nropv1 "github.com/openshift-kni/numaresources-operator/api/numaresourcesoperator/v1"
+	"github.com/openshift-kni/numaresources-operator/internal/remoteexec"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	e2etestenv "github.com/k8stopologyawareschedwg/resource-topology-exporter/test/e2e/utils/testenv"
+	e2eclient "github.com/openshift-kni/numaresources-operator/test/utils/clients"
+	"github.com/openshift-kni/numaresources-operator/test/utils/deploy"
+	"github.com/openshift-kni/numaresources-operator/test/utils/objects"
+)
+
+// This test verifies that the HTTPS metrics endpoints, are accessible and serving metrics.
+var _ = Describe("metrics exposed securely", Serial, func() {
+	metricsAddress := "127.0.0.1"
+	var namespace string
+	var err error
+	var nropObj *nropv1.NUMAResourcesOperator
+	var initialized bool
+
+	BeforeEach(func() {
+		if initialized {
+			return
+		}
+		nropObj = objects.TestNRO()
+		nname := client.ObjectKeyFromObject(nropObj)
+
+		err = e2eclient.Client.Get(context.TODO(), nname, nropObj)
+		Expect(err).NotTo(HaveOccurred(), "failed to get the NRO resource: %v", err)
+
+		Expect(len(nropObj.Status.NodeGroups)).To(BeNumerically(">=", 1), "node groups not reported, nothing to do")
```

Comment:
simpler:
`Expect(nropObj.Status.NodeGroups).ToNot(BeEmpty()), "node groups not reported, nothing to do")`